packages/calcite-components/src/components/swatch/swatch.scss

```diff
@@ -0,0 +1,122 @@
+/**
+ * CSS Custom Properties
+ *
+ * These properties can be overridden using the component's tag as selector.
+ *
+ * @prop --calcite-swatch-corner-radius: Specifies the component's corner radius.
+ *
+ */
+
+// AUTO-GENERATED — do not modify. Changes will be overwritten.
+//
+// Internal CSS custom properties for component use only. Overwriting is not recommended.
+//
+// --calcite-internal-swatch-inset
+// --calcite-internal-swatch-size
+
+:host {
+  @apply block;
+  --calcite-internal-swatch-inset: var(--calcite-spacing-xxs);
+}
+
+:host([scale="s"]) {
+  .container {
+    --calcite-internal-swatch-size: var(--calcite-spacing-xl);
+  }
+}
+
+:host([scale="m"]) {
+  .container {
+    --calcite-internal-swatch-size: var(--calcite-spacing-xxl);
+  }
+}
+
+:host([scale="l"]) {
+  .container {
+    --calcite-internal-swatch-size: var(--calcite-spacing-xxxl);
+  }
+}
+
+.container {
+  @apply relative focus-base justify-center box-border  overflow-hidden;
+  font-size: var(--calcite-internal-swatch-font-size, var(--calcite-font-size));
+  block-size: var(--calcite-internal-swatch-size, auto);
+  inline-size: var(--calcite-internal-swatch-size, auto);
+  min-inline-size: var(--calcite-internal-swatch-size, auto);
+  border-radius: var(--calcite-swatch-corner-radius, 0);
+
+  &:not(.non-interactive):hover {
+    @apply cursor-pointer;
+  }
+  &:not(.non-interactive):hover {
+    box-shadow: 0 0 0 2px var(--calcite-color-border-2);
```

Comment:
should this use a shadow css var?

Reply:
Yeah we don't have a semantic show token for this. We could use `var(--calcite-border-width-md)` for the "2px" if we want to.